modules/core/src/views/view.js

```diff
@@ -80,7 +80,7 @@ export default class View {
 
     // Resolve relative viewport dimensions
     const viewportDimensions = this.getDimensions({width, height});
-    const props = Object.assign({viewState}, viewState, this.props, viewportDimensions);
+    const props = Object.assign({}, viewState, this.props, viewportDimensions);
```

Comment:
Hmm, I'd love to see the typescript for this kind of props merging...

Just a sense that moving towards more strictly typed props and less "scripty/sloppy" merging of unrelated objects might be good for maintainability, this looks like potentially a small step backwards. 

But as long as it is for a good reason.